### PR TITLE
Plugin system documentation

### PR DESCRIPTION
**Pull request by rete**
BEGINRELEASENOTES
- Implemeted plugin system documentation
- Added CSS stylesheet with warning, info, success and error bloc style definition 
ENDRELEASENOTES

### CODE REVIEW

File: docs/user-manual/plugin-system.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+
+# The plugin system in a nutshell
+
+The plugin system component of the DQM4hep core package has a central place in the software design. The genericity of DQM4hep relies on plugins, that can be loaded at runtime. Parts of the framework can thus easily changed at runtime by specifying a different set of plugins to use.
+
```
Comment (apingault) [inline docs/user-manual/plugin-system.md]: I would remove the comma in the second sentence. 
Missing a word in the next sentence : Parts of the framework can thus __be__ easily changed ...

File: docs/user-manual/plugin-system.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+<div class="info-msg">
+  <i class="fa fa-info-circle"></i>
+  Most of the binaries provided the DQM4hep packages load the shared libraries using the third option, with the environment variable <span style="font-style: italic">DQM4HEP_PLUGIN_DLL</span>, at the beginning of the main. 
+</div>
+
+Before running a DQM4hep program it is usually a good idea to export <span style="font-style: italic">DQM4HEP_PLUGIN_DLL</span> with your plugin libraries inside :
+
+```shell
+export DQM4HEP_PLUGIN_DLL=libSpiderman.so:libThor.so:libIronman.so
```
Comment (apingault) [inline docs/user-manual/plugin-system.md:91]: Why are you mixing DC and Marvel characters?

File: docs/user-manual/plugin-system.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+or with your own libraries to see what are the compiled plugins :
+
+```shell
+# show our two 'shape' plugins
+export DQM4HEP_PLUGIN_DLL=libShapePlugins.so
+dqm4hep-dump-plugins
+```
+
```
Comment (apingault) [inline docs/user-manual/plugin-system.md]: Maybe add the output of the command line here

**Review by apingault — DISMISSED** [review]

File: docs/user-manual/plugin-system.md (hunk 1 above)
Comment (rete) [reply docs/user-manual/plugin-system.md]: Habile !

File: docs/user-manual/plugin-system.md (hunk 2 above)
Comment (rete) [reply docs/user-manual/plugin-system.md:91]: Super heroes are super heroes. They save the world, that's all ...

File: docs/user-manual/plugin-system.md (hunk 3 above)
Comment (rete) [reply docs/user-manual/plugin-system.md]: Will be added yes

**Pull request merged**